app/hooks/useLiquidityPosition.ts

```diff
@@ -217,6 +229,12 @@ export function useLiquidityPosition({
           });
         }
 
+        // Handle Cosmos chain transactions
+        if (wallet.chainType === ChainKey.GAIACHAIN) {
+          const cosmosAmount = amount * 10 ** parseInt(pool.nativeDecimal);
```

Comment:
I think it would be better to use Xchain's functions 
- baseAmount
- assetAmount